server/src/main/java/io/crate/expression/reference/doc/lucene/DocCollectorExpression.java

```diff
@@ -85,8 +85,11 @@ private ChildDocCollectorExpression(Reference ref) {
 
         @Override
         public Object value() {
-            // correct type detection is ensured by the source parser
-            return source.get(ref.column().path());
+            try {
+                return ref.valueType().sanitizeValue(source.get(ref.column().path()));
```

Comment:
Instead of always sanitizing the value, we could only do this if any parent of of type `OBJECT(IGNORED` but we'd need to resolve the parent's reference to do that. As a reference can only refer to it's parent `ColumnIdent`, we'd need to inject some reference resolving logic here. Is this worth the effort or are we fine with always sanitizing the value? I tend to the later.

Reply:
Looks like we could pass the `ReferenceTree` from the `DocTableInfo` to the `LuceneReferenceResolver`. The DocTableInfo is already available everywhere we currently construct a `LuceneReferenceResolver`.

If that works I'd prefer doing it limited to `object(ignored)` to avoid ending up hiding potential bugs for other scenarios, but if it's more effort than it looks like on a glance I guess I'd also tend to always sanitize over adding more complexity.

Reply:
I've added this logic with https://github.com/crate/crate/pull/16930/commits/431d277992527ee707660acf9cbff6113b4c728b.